### PR TITLE
fix(rocklet): use cgroup metrics for container CPU instead of psutil (#945)

### LINKED ISSUE #945: [Bug] Rocklet CPU utilization reports host-level metrics instead of container-level
Description
             
  LinuxRocklet.get_statistics() uses psutil.cpu_percent() to report CPU utilization. However, psutil reads from /proc/stat, which reflects host-level CPU usage rather
  than the container's cgroup-constrained resources. This causes two problems:                                                                                         
  
  1. Inaccurate readings — On a multi-tenant host, a container allocated 2 cores reports the average utilization of the entire 64-core machine, significantly          
  understating its actual load.                             
  2. Quota throttling is invisible — When a container hits its CFS quota limit and gets throttled, psutil may still report low utilization, misleading scheduling      
  decisions.                                                                                                                                                           
  
  Expected Behavior                                                                                                                                                    
                                                            
  Rocklet should read cgroup (v1/v2) CPU accounting files (cpuacct.usage / cpu.stat) and quota settings (cpu.cfs_quota_us / cpu.max) to compute CPU usage as a         
  percentage of the container's allocated quota.
                                                                                                                                                                       
  Solution                                                  

  - Add rock/utils/cgroup_stats.py module with cgroup v1 and v2 support:                                                                                               
    - Read cgroup CPU usage in nanoseconds
    - Read container CPU quota (quota/period)                                                                                                                          
    - Compute delta-based CPU percentage between successive calls
  - Gracefully fall back to psutil.cpu_percent() when cgroup files are unavailable (non-container environments)                                                        
  - Replace psutil.cpu_percent() with cgroup_cpu_percent() in LinuxRocklet.get_statistics()                                                                            
                                                                                                                                                                       
  Impact                                                                                                                                                               
                                                                                                                                                                       
  Affects all CPU metrics reported by rocklet. After the fix, admin scheduling and autoscaling decisions will be based on accurate container-level CPU data.

### PR DESCRIPTION
**Pull request by jake11-oho**
## Summary                                                                                                                                                          
                                                                                                                                                                       
   - Replace `psutil.cpu_percent()` with cgroup-aware CPU measurement in `LinuxRocklet.get_statistics()` to report accurate container-level utilization instead of     
   host-level metrics                                                                                                                                                  
   - Add `rock/utils/cgroup_stats.py` supporting both cgroup v1 and v2 with automatic fallback to psutil in non-container environments
   - Add comprehensive unit tests covering v1/v2 detection, CPU calculation, quota parsing, and edge cases

   Fixes #945

   ## Test Plan

   - [x] `uv run pytest tests/unit/utils/test_cgroup_stats.py` — all unit tests pass
   - [x] `uv run pytest -m "not need_ray and not need_admin and not need_admin_and_network" --reruns 1` — no regressions in fast tests
   - [ ] Deploy rocklet in a cgroup v2 container and verify `GET /statistics` returns CPU % relative to container quota
   - [x] Deploy rocklet in a cgroup v1 container and verify correct CPU readings
   - [x] Run rocklet outside a container (bare metal / macOS) and verify fallback to psutil works

   🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: rock/utils/cgroup_stats.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import psutil
+
+_last_cpu_usage: int | None = None
+_last_cpu_time: int | None = None
+
+_cgroup_version: int | None = None
+_cpu_quota: float | None = None
+
+
+def _detect_cgroup_version() -> int:
```
Comment (StephenRi) [inline rock/utils/cgroup_stats.py]: 改成面向对象

Comment (jake11-oho) [reply rock/utils/cgroup_stats.py]: fixed

**Review by StephenRi — APPROVED** [review]
